src/util.ts

```diff
@@ -92,3 +92,19 @@ export function clearBusyState(...elements: Element[]) {
     element.removeAttribute("aria-busy")
   }
 }
+
+export function waitForLoad(element: HTMLLinkElement): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const onLoad = () => {
+      element.removeEventListener("error", onError)
+      resolve()
+    }
+    const onError = () => {
+      element.removeEventListener("load", onLoad)
+      reject()
```

Comment:
I had suggested `reject()` here to handle an `error`. After thinking about it a little longer, I wonder if `resolve()` is a better response. I suspect that `reject()` here would halt the other Promises, which feels like an overreaction to an external resource failing to load.

If our intent here is only to synchronize, I think `resolve()` would be less brittle:

```typescript
return new Promise((resolve) => {
  const onComplete = () => {
    element.removeEventListener("load", onComplete)
    element.removeEventListener("error", onComplete)
    resolve()
  }
  element.addEventListener("load", onComplete)
  element.addEventListener("error", onComplete)
})
```